artmanflow/web/sources_generation.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from flask import Blueprint, Response, \
+    render_template, request, redirect, send_from_directory
+
+from artmanflow.steps.sources_generation_host import SourcesGenerationHost
+from artmanflow.steps.common import ConfigUtils
+from artmanflow.web.html_utils import HtmlUtils
+from artmanflow.web.service_utils import ServiceUtils
+
+src_gen = Blueprint('sources_generation', __name__,
+                    url_prefix='/sources-generation')
+
+
+@src_gen.route('/new')
+def sources_generation_new():
+    config = _params_from_yaml(
+        ServiceUtils.get_step_default_config('sources_generation.yaml'))
+    return render_template('sources_generation_new.html', config=config)
+
+
+@src_gen.route('', methods=['POST'])
+def sources_generation():
+    config_yaml = ServiceUtils.get_step_config(request.form,
+                                               'sources_generation.yaml',
+                                               _params_to_yaml)
+
+    execution_id = ConfigUtils.generate_id('src-gen-')
+    config_yaml['execution_id'] = execution_id
+    step = SourcesGenerationHost(config_yaml)
+    ServiceUtils.run_host_step(step)
+
+    return redirect("/sources-generation/%s" % execution_id)
+
+
+@src_gen.route('/<execution_id>')
```

Comment:
Force this to be an int or a UUID (whichever you need); right now this is an XSS vulnerability.

Reply:
Also literally anything uncaught by another route will route here.

Reply:
This is under 'sources_generation' blueprint, which means that this will match anything with that prefix-only, which is fine. The routing principles were taken from [Ruby-on-Rails](http://guides.rubyonrails.org/routing.html#crud-verbs-and-actions), which, I believe, has pretty decent default routing structure.